File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantMethodTypeInfo.java
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_MethodType_info {
    u1 tag;
    u2 descriptor_index;
}
*/
public class ConstantMethodTypeInfo extends ConstantInfo {

    {
        u2("descriptor_index");
    }
    
    @Override
    protected String loadDesc(ConstantPool pool) {
        int descriptorIndex = super.getUInt("descriptor_index");
        return pool.getUtf8String(descriptorIndex);
    }
    
}

File: src/main/kotlin/com/github/zxh/classpy/classfile/constant/ConstantMethodTypeInfo.kt
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_MethodType_info {
    u1 tag;
    u2 descriptor_index;
}
*/
class ConstantMethodTypeInfo : ConstantInfo() {

    init {
        u2("descriptor_index");
    }
    
    override fun loadDesc(cp: ConstantPool): String {
        val descriptorIndex = super.getInt("descriptor_index");
        return cp.getUtf8String(descriptorIndex);
    }
    
}
